lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -69,6 +69,119 @@ const OptTable * hlsl::options::getHlslOptTable() {
   return g_HlslOptTable;
 }
 
+static std::vector<ArgPair> ComputeArgPairsFromArgList(const llvm::opt::InputArgList &argList) {
```

Comment:
Rather than returning a std::vector which guarantees a copy, can we pass a `SmallVectorImpl<>&` to avoid copying the array on return? That also allows us to avoid resizes.